src/printer.js

```diff
@@ -621,7 +621,40 @@ function genericPrintNoParens(path, options, print) {
         } else {
           parts.push(printPropertyKey(path, print));
         }
-        parts.push(": ", path.call(print, "value"));
+        let noBreak = [
+          "ArrayExpression",
+          "ArrayPattern",
+          "ArrowFunctionExpression",
+          "AssignmentPattern",
+          "CallExpression",
+          "FunctionExpression",
+          "Identifier",
+          "Literal",
+          "LogicalExpression",
```

Comment:
How come logical expression is here but binary expression isn't? Should probably do the same for both